### ISSUE TITLE
 `Cannot read property 'toString' of undefined` error on line 93 of Util.js.

### ISSUE THREAD

**Issue by yogesh-MS** [microsoft/typed-rest-client#272]
I get a `Cannot read property 'toString' of undefined` error on line 93 of Util.js. 
 
`resolve(buffer.toString(charset || 'utf-8'));`

The buffer is undefined when running Chrome 86.0.4240.198 in debug mode with the following command line:

> sudo /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --remote-debugging-port=9222 --disable-web-security --user-data-dir=$HOME/chromeTemp

If I remove `--disable-web-security --user-data-dir=$HOME/chromeTemp` from the command line, I get more cryptic error message:

`Failed to fetch `

Any ideas why the buffer would be undefined? I would propose a check on undefined before calling toString().

I'm using version typed-rest-client@1.8.0 inside the project https://github.com/microsoft/azure-devops-node-api.

_Originally posted by @lpcarignan in https://github.com/microsoft/typed-rest-client/issues/189#issuecomment-728167073_

**Comment by anatolybolshakov** [comment]
Hi @YogeshIITK looks like zlib.gunzip returns undefined result without any error - it's definitely makes sense to add null check then - let us take a look.

**Comment by YogeshIITK** [comment]
I think- Decompression is already done by browser.. and This function again trying to decompress the responses content... that's why buffer becomes undefined.. Earlier our app code was working fine with "typed-rest-client": "1.5.0" but when we took upgrade to  "typed-rest-client": "1.8.4",, we are seeing this issue..

**Comment by kuleshovilya** [comment]
Don't think that's the case here, client's execution won't be affected by that, or at least shouldn't be. And we tested this exact behavior on mock react app, and it worked fine. We'll look more into what can be done, maybe if it's possible you could at least provide some abstract repro steps for that, so we could repeat your situation more closely. Also, on related note - what NodeJS version are you using?

**Comment by kuleshovilya** [comment]
Yeah, after extensive testing I'm pretty sure it's not the browser\decoding issue. The value comes into client either encoded or decoded, and then processed accordingly, code handles it correctly. So, we would need some context to troubleshoot the issue. Let's start with:
NodeJS version
Could you please describe context in which this issue occurs? As in conditions, what it's supposed to do etc

**Comment by kuleshovilya** [comment]
Any update on the request?

**Comment by kuleshovilya** [comment]
I'll close the ticket for now, but you can always reopen it or ping me at v-ikuleshov@microsoft.com

**Comment by yogesh-MS** [comment]
![image](https://user-images.githubusercontent.com/3176607/133422588-a737c1ee-dace-4c21-8c02-42c21a349648.png)

**Comment by kuleshovilya** [comment]
@yogesh-MS Rechecked the function and still, it works fine with gzip content in chrome or any other browser, it doesn't re-decompress it, since it first checks for the content, whether its gzip or not. 
Can you please check whether the correct value comes in to the client from the app side? 
The older version, which worked for you, didn't have value checks the newer one has, so that might be the reason for it not working now.

**Comment by kuleshovilya** [comment]
@yogesh-MS Any update on the request?

**Comment by anatolybolshakov** [comment]
Hey @yogesh-MS there seems to be an issue with NodeJS not relevant to the typed-restl-client library itself - there is an empty buffer parameter in [line](https://github.com/microsoft/typed-rest-client/blob/master/lib/Util.ts#L96) for some reason coming in callback for zlib.gunzip method - although we did not found any relevant known NodeJS issues for this...
Could you please open a ticket in NodeJS [repo](https://github.com/nodejs/node) for further trhoubleshooting?
